nixos/modules/virtualisation/linode-config.nix

```
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+with lib;
+{
+  imports = [ ../profiles/qemu-guest.nix ];
+
+  ### Allow Root logins only using SSH keys
+  services.openssh = {
+    enable = true;
+
+    permitRootLogin = "prohibit-password";
+    passwordAuthentication = mkDefault false;
+  };
+
+  ### Set networking config
+  #
+  networking = {
+    usePredictableInterfaceNames = false;
+    useDHCP = false;
+    interfaces.eth0.useDHCP = true;
```

It looks like Linode expects IPv6 privacy extensions to be off by default (https://www.linode.com/docs/guides/manual-network-configuration/#static-vs-dynamic-addressing) so you might actually also want to add this:

```suggestion
    interfaces.eth0 = {
      useDHCP = true;
      tempAddress = "disabled";
    };
```

---

Tested and implemented in 146dbfbf482